### DISCUSSION TITLE
VS Code + FORTRAN IntelliSense

### DISCUSSION THREAD

**Discussion by lukhirt** [fortran-lang/fortls#65] [Q&A] [answered]
Hi,

I just learned that Chris Hansen's repo (fortran-language-server) is no longer under development and you are the one who keeps it going nowadays :)
Currently I am still using the setup with Chris Hansen's fortls version, simply because I am making use of his VS Code extension (FORTRAN IntelliSense). I was thus wondering if your version of the fortran server may be used instead for this VS Code extension.
If not, could we please discuss shortly how could I benefit from your version of fortls when working in VS Code? :)

Cheers,
Lukasz

**Comment by gnikit** [comment, marked as answer]
Hi @lukhirt,

To briefly answer your question, yes you can replace `fortran-language-server` with `fortls`. This is the main reason why I kept the name `fortls` so people and code editor extensions would not have to support 2 language servers.

To be a bit more detailed:

Simply uninstalling `fortran-language-server` and installing `fortls` should already give you access to most new features I have added, but it will certainly not give you access, through VS Code `settings.json`, to all the new settings. You can use a `.fortls` file to circumvent that, since I have unified the command-line and `json` interfaces, but there is a **much easier** way.

I am also a developer for the VS Code [Modern Fortran extension](https://github.com/krvajal/vscode-fortran-support/), so in VS Code we have complete control over Fortran support and features. I made the decision a while back that relying on `FORTRAN Intellisense` to have Language Server features in VS Code was a really bad idea, since that extension hasn't been updated for years.

In the Pre-Release version of the extension, and all versions moving forward, we can natively interface with `fortls`, so you no longer need to have `FORTRAN Intellisense` installed in your system.

A range of new settings should be available under `fortran.fortls.` to configure the language server from within Modern Fortran. FYI this still a work in progress see https://github.com/krvajal/vscode-fortran-support/pull/293

The other main reason why I recommend to individuals to switch is for the shear number of bug fixes and new features `fortls` has when compared to `fortran-language-server`. You can find a condensed summary [here](https://gnikit.github.io/fortls/fortls_changes.html)

I hope that clears things up a bit.

P.S. the default settings Modern Fortran sets to `fortls` are pretty reasonable so it should be plug-n-play in most cases. Let me know if you have any troubles setting up.

**Reply by lukhirt** [reply]
Hi @gnikit,

Wow! That's a great news that everything is under Modern Fortran extension now.
I just gave it a try and yes, it works - I had to remove the other extension though to make it work, but well...
I will need to dive into documentation now and see what's in there, can't wait!

By the way, thanks for a thorough description, that helped a lot.

**Reply by gnikit** [reply]
No worries, glad I could help
